reactfront/src/components/graph-utils.js

```diff
@@ -24,16 +24,16 @@ function roundToSigFigs(x, sigFigs) {
 
 // Formatted label including SI unit prefixes
 export function formatUnit(val, unit, sigFigs = 3) {
-  return Math.abs(val) >= 1_000_000_000_000  ? `${roundToSigFigs(val / 1_000_000_000_000, sigFigs)} T${unit}`
-       : Math.abs(val) >= 1_000_000_000      ? `${roundToSigFigs(val / 1_000_000_000,     sigFigs)} G${unit}`
-       : Math.abs(val) >= 1_000_000          ? `${roundToSigFigs(val / 1_000_000,         sigFigs)} M${unit}`
-       : Math.abs(val) >= 1_000              ? `${roundToSigFigs(val / 1_000,             sigFigs)} k${unit}`
-       : Math.abs(val) >= 1                  ? `${roundToSigFigs(val,                     sigFigs)} ${unit}`
-       : Math.abs(val) >= 0.001              ? `${roundToSigFigs(val * 1_000,             sigFigs)} m${unit}`
-       : Math.abs(val) >= 0.000_001          ? `${roundToSigFigs(val * 1_000_000,         sigFigs)} μ${unit}`
-       : Math.abs(val) >= 0.000_000_001      ? `${roundToSigFigs(val * 1_000_000_000,     sigFigs)} n${unit}`
-       : Math.abs(val) >= 0.000_000_000_001  ? `${roundToSigFigs(val * 1_000_000_000_000, sigFigs)} p${unit}`
-       : `0 ${unit}`
+  let prefixes = ["T", "G", "M", "K", "", "m", "μ", "n", "p"]
+  let trill = Math.pow(1000, 4) // 1 trillion for T
+  for (let i = 0; i < prefixes.length; i++) {
+      let divider = trill / Math.pow(1000, i)
+      if (Math.abs(val) >= divider) {
+          return roundToSigFigs(val / divider, sigFigs) + prefixes[i] + unit
+      }
+  }
+  // was "return '0 ' + unit" but I changed it to:
+  return roundToSigFigs(val, sigFigs) + unit
```

Comment:
* Could you add spaces between the number and the prefix+unit? Eg `formatUnit(0.0015, 'V') === "1.5 mV"`
* For really small numbers, it prints `formatUnit(0.00000000000001, 'V') == "1e-16V"` when we probably want `0.0001 pV`. The old version is much worse here for what it's worth, it just says `0 V` when it's less than 1 picovolt
* Instead of defining `trill`, could we use `const divider = Math.pow(10, something)` and maybe store tuples of `(prefix, power_of_10)` in `prefixes` to use for `something`?
* Comment at the end looks redundant